### PR TITLE
CLI metrics changes

### PR DESCRIPTION
**Pull request by shamantraghav**
# GitHub, NPM Metrics Submission #

Submits metrics to Datadog via GitHub action
### Checklist
- [x] I acknowledge that all my contributions will be made under the project's license
- [ ] I have made a material change to the repo (functionality, testing, spelling, grammar)
- [ ] I have read the [Contribution Guidelines](https://github.com/twilio/twilio-cli/blob/main/CONTRIBUTING.md) and my PR follows them
- [ ] I have titled the PR appropriately
- [ ] I have updated my branch with the main branch
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added the necessary documentation about the functionality in the appropriate .md file
- [ ] I have added inline documentation to the code I modified

If you have questions, please file a [support ticket](https://twilio.com/help/contact), or create a GitHub Issue in this repository.

### CODE REVIEW

File: .github/scripts/github_metrics.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
+"""
+Submit metrics returns "Payload accepted" response
+"""
+
```
Comment (shamantraghav) [inline .github/scripts/github_metrics.py:4]: For reference: https://docs.datadoghq.com/api/latest/metrics/

File: .github/workflows/npm-download-metrics.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
+name: NPM Download Metrics
```
Comment (shamantraghav) [inline .github/workflows/npm-download-metrics.yml:1]: Successful sample run: https://github.com/shamantraghav/twilio-cli/actions/runs/1704804569

File: .github/scripts/npm_metrics.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+npm_url="https://api.npmjs.org/downloads/point/"+yesterday_url_format+"/twilio-cli"
+
+response = requests.get(npm_url)
+download_count=response.json()
+download_count_float=float(download_count['downloads'])
+
+body = MetricsPayload(
+    series=[
+        Series(
+            metric="npm.download.count",
```
Comment (onuzbee) [inline .github/scripts/npm_metrics.py:26]: Is it a common Datadog bucket for metrics or are these metric names in global namespace for Twilio org? 
If it's global- I'd suggest naming the metrics with a twilio-cli prefix.

File: .github/workflows/npm-download-metrics.yml [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+          python-version: 3.8 
+      - name: Execute py script 
+        env:
+          DD_SITE: datadoghq.com
+          DD_API_KEY: ${{ secrets.DATADOG_API_KEY }}   
+        run: |
+          pip3 install datadog-api-client
+          pip3 install requests
+          pip3 install datetime
+          python3 .github/scripts/npm_metrics.py
```
Comment (onuzbee) [inline .github/workflows/npm-download-metrics.yml:27]: Can we club all metric related files to a common directory structure:
```
.github/scripts/metrics/[]
.github/workflows/metrics/[] // If github can parse from a subdir
```

File: .github/scripts/github_metrics.py [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+from datadog_api_client.v1.model.series import Series
+import requests
+
+
+github_url="https://api.github.com/repos/twilio/twilio-cli/releases"
+response = requests.get(github_url)
+output=response.json()
+rpm_count = deb_count = exe_count = pkg_count = 0
+
+for release in output:
```
Comment (shrutiburman) [inline .github/scripts/github_metrics.py:20]: Does this mean that the output json will have all releases in response and this loop would go through all of them, each time?

File: .github/scripts/github_metrics.py [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+        if(platform_exe['name'][-3:]=="rpm"):
+            rpm_count+=platform_exe['download_count']
+        elif(platform_exe['name'][-3:]=="deb"):
+            deb_count+=platform_exe['download_count']
+        elif(platform_exe['name'][-3:]=="exe"):
+            exe_count+=platform_exe['download_count']
+        elif(platform_exe['name'][-3:]=="pkg"):
+            pkg_count+=platform_exe['download_count']
+
+download_count_rmp_float=float(rpm_count)
```
Comment (shrutiburman) [inline .github/scripts/github_metrics.py:31]: [nit] : download_count_**rpm**_float

File: .github/scripts/github_metrics.py [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+download_count_pkg_float=float(pkg_count)
+
+
+body_rpm = MetricsPayload(
+    series=[
+        Series(
+            metric="github.rpm.downloads_count",
+            type="count",
+            points=[Point([datetime.now().timestamp(), download_count_rmp_float])],
+            tags=["test:github"],
```
Comment (shrutiburman) [inline .github/scripts/github_metrics.py:43]: Would the tag be manually added anywhere or is this where we define the tag? ["test:github"]

File: .github/scripts/github_metrics.py [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+        )
+    ]
+)
+
+configuration = Configuration()
+with ApiClient(configuration) as api_client:
+    api_instance = MetricsApi(api_client)
+    response1 = api_instance.submit_metrics(body=body_rpm)
+    print("Github rpm metrics submission status: ",response1)
```
Comment (shrutiburman) [inline .github/scripts/github_metrics.py:85]: Is there a log generated with these statements or is there a sample response that we can see?

File: .github/scripts/github_metrics.py (hunk 5 above)
Comment (shamantraghav) [reply .github/scripts/github_metrics.py:20]: It would have the data for the last 30 releases

File: .github/scripts/github_metrics.py (hunk 7 above)
Comment (shamantraghav) [reply .github/scripts/github_metrics.py:43]: We define it here. These tags can be used while plotting the metrics

File: .github/scripts/github_metrics.py (hunk 8 above)
Comment (shamantraghav) [reply .github/scripts/github_metrics.py:85]: These logs just make it easier for us to debug and find out which step failed. When successful, the response is just a 200 status. These logs will just help us save some time

File: .github/workflows/npm-download-metrics.yml (hunk 4 above)
Comment (shamantraghav) [reply .github/workflows/npm-download-metrics.yml:27]: Sure

File: .github/scripts/npm_metrics.py (hunk 3 above)
Comment (shamantraghav) [reply .github/scripts/npm_metrics.py:26]: Should be a separate one as we have a unique API key. The others won't be able to see this

File: .github/scripts/github_metrics.py (hunk 5 above)
Comment (shrutiburman) [reply .github/scripts/github_metrics.py:20]: Okay. This is the data we will use to plot the graph then?

File: .github/scripts/github_metrics.py (hunk 8 above)
Comment (shrutiburman) [reply .github/scripts/github_metrics.py:85]: Any sample logs available?

File: .github/scripts/github_metrics.py (hunk 1 above)
Comment (shrutiburman) [reply .github/scripts/github_metrics.py:4]: I see there's a new secret that we would be adding `secrets.DATADOG_API_KEY`, any other required changes in existing permissions?

Comment (shamantraghav) [reply .github/scripts/github_metrics.py]: Nope, just this key

File: .github/scripts/github_metrics.py (hunk 5 above)
Comment (shamantraghav) [reply .github/scripts/github_metrics.py]: Yes

File: .github/scripts/github_metrics.py (hunk 8 above)
Comment (shamantraghav) [reply .github/scripts/github_metrics.py]: https://github.com/shamantraghav/twilio-cli/actions/runs/1704804569

Comment (shrutiburman) [reply .github/scripts/github_metrics.py]: Okay, same as the one mentioned earlier.👍

**Review by shrutiburman — APPROVED** [review]

**Review by onuzbee — APPROVED** [review]
Neat work! 👏🏻

**Pull request merged**